### DISCUSSION TITLE
Update items inside BagPart

### DISCUSSION THREAD

**Discussion by EricBui619** [OrchardCMS/OrchardCore#9434] [Q&A]
Hi all, I have a problem when using BagPart. I created content type with a BagPart that contains content items of other my defined content type. So the problem is when I perform any CRUD operation in bag part, OC will create a new document instead of modifying the current one. This results in you will lost 
data if you don't pass all items of bag part when calling api.

For example: I have a content type with a BagPart that are containing 3 items. Let assume they have ID ordered 1, 2, 3. If I only want to update item with id=2, I still have to pass all 3 items when calling api/content to update content items. If I pass just item with id=2, OC will create a new document with containing only id=2 in its bag part, id=1 and 3 will lost

So, how can I perform CRUD in this scenario exactly?

**Comment by deanmarcussen** [comment]
have a read of this discussion, it might help
https://github.com/OrchardCMS/OrchardCore/discussions/6927

Basically a bagpart is contained inside another json document, so you need to retrieve that, then modify the bagpart items as you wish.

If you just want to do single updates to the item inside it, you'll need to write your own api controller to handle the scenario as you wish.

**Reply by EricBui619** [reply]
@deanmarcussen  Thanks for your support. I'll try it
